python/ikarus/utils/utils.hh

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include <dune/python/pybind11/eigen.h>
+#include <dune/python/pybind11/operators.h>
+#include <dune/python/pybind11/pybind11.h>
+
+#include <ikarus/utils/tensorutils.hh>
+
+void addBindingsToUtils() {
+  namespace py = pybind11;
+  using namespace pybind11::literals;
+  using namespace Ikarus;
+
+  auto utils = pybind11::module::import("ikarus.utils");
+
+  /**
+   * \brief Converts a square 1x1, 2x2 or 3x3 matrix to a Voigt notation vector.
+   *  \ingroup tensor
+   * \param E Input matrix of size (size x size).
+   * \param isStrain Flag indicating whether the conversion is for strain (true) or not (false) (default is true)..
+   * \return  Vector with components in Voigt notation vector.
+   *
+   * This function converts a square matrix to a Voigt notation vector, which contains the unique components of
+   * the input matrix.
+   *
+   * The optional isStrain parameter allows the user to specify whether the conversion is intended for strain
+   * calculations. If isStrain is true, the off-diagonal components are multiplied by 2, providing the correct Voigt
+   * notation for symmetric strain tensors.
+   */
+  utils.def(
+      "toVoigt",
+      [](Eigen::MatrixXd mat, bool isStrain = true) {
```

Comment:
This could be simplified using the variable but fixed sized Eigen types as:

```cpp
utils.def(
      "toVoigt",
      [](Eigen::Matrix<double,Eigen::Dynamic, Eigen::Dynamic,0,3,3> mat, bool isStrain = true) {
       return toVoigt(m, isStrain);
      },
      py::arg("matrix"), py::arg("isStrain") = true);

with a few modifications to `toVoigt`, which supports hes hybrid matrices.
```
See https://godbolt.org/z/nEvrWszfz for reference